src/state/slices/opportunitiesSlice/selectors.ts

```diff
@@ -0,0 +1,75 @@
+import { createSelector } from '@reduxjs/toolkit'
+import type { ReduxState } from 'state/reducer'
+import { selectAccountIdParamFromFilter, selectUserStakingIdParamFromFilter } from 'state/selectors'
+
+import type {
+  LpId,
+  OpportunityMetadata,
+  StakingId,
+  UserStakingOpportunity,
+} from './opportunitiesSlice'
+import { deserializeUserStakingId } from './utils'
+
+// IDs selectors
+export const selectLpIds = (state: ReduxState) => state.opportunities.lp.ids
+export const selectStakingIds = (state: ReduxState) => state.opportunities.staking.ids
+export const selectUserStakingIds = (state: ReduxState) => state.opportunities.userStaking.ids
+
+export const selectLpOpportunitiesByAccountId = (state: ReduxState) =>
+  state.opportunities.lp.byAccountId
+export const selectStakingOpportunitiesByAccountId = (state: ReduxState) =>
+  state.opportunities.staking.byAccountId
+export const selectUserStakingOpportunitiesById = (state: ReduxState) =>
+  state.opportunities.userStaking.byId
+export const selectStakingOpportunitiesById = (state: ReduxState) =>
+  state.opportunities.staking.byId
+
+// "Give me all the LP opportunities this AccountId has", so I can get their metadata from the slice, and then their data from the portfolio slice
+export const selectLpOpportunityIdsByAccountId = createSelector(
+  selectLpOpportunitiesByAccountId,
+  selectAccountIdParamFromFilter,
+  (lpIdsByAccountId, accountId): LpId[] => lpIdsByAccountId[accountId] ?? [],
+)
+
+// "Give me all the staking opportunities this AccountId has", so I can get their metadata and their data from the slice
+export const selectStakingOpportunityIdsByAccountId = createSelector(
+  selectStakingOpportunitiesByAccountId,
+  selectAccountIdParamFromFilter,
+  (stakingIdsByAccountId, accountId): StakingId[] => stakingIdsByAccountId[accountId] ?? [],
+)
+
+// I'm not a selector buddy, don't prepend me with `select`
+export const deserializeStakingIdFromUserStakingId = createSelector(
```

Comment:
dont export me from selectors file - move me to utils or reselect dev tools will be sad

Reply:
Technically a selector, taking state as input, just not returning any state field, so I reckon we should leave it here

Reply:
ah ok sure, it is actually a selector

Reply:
If the intent isn't clear, let's add the `select` prefix, I initially felt like `select` carries the mental model of selecting from the state, but we do have selectors that `selectXYZParamFromFilter` which is effectively the same - what about `selectDeserializeStakingIdFromUserStakingIdParam` ?

Reply:
04cff1989c0782e1424043625cd497c425a6670a